prometheus_api_client/metric_range_df.py

```diff
@@ -74,4 +78,8 @@ def __init__(
             data=row_data, index=index, columns=columns, dtype=dtype, copy=copy
         )
 
+        # convert to DateTime type instead of Float64
+        if ts_floats_to_datetime:
+            self["timestamp"] = to_datetime(self["timestamp"], unit="s")
```

Comment:
This is extremely slow, 100× slower than using `datetime.datetime.fromtimestamp()`. You can verify this yourself using [this short perf comparison code](https://gist.github.com/mmazur/c813d12b7e01ad24271e5ba8852294d7).

Reply:
Hmm so it seems to me like the perf comparison code you shared above uses `pd.to_datetime` slightly differently than how it's being used in this PR. When I [pass the `pd.Series` itself as an arg](https://gist.github.com/chauhankaranraj/f1eb33a990090cbd77af1c908366e312#file-to_datetime_vs_fromtimestamp-py-L15) instead of [passing individual values through `.apply` and `lambda`](https://gist.github.com/mmazur/c813d12b7e01ad24271e5ba8852294d7#file-to_datetime_vs_fromtimestamp-py-L15), I actually see a performance gain of ~30x over using `datetime.datetime.fromtimestamp()`.

Does that seem correct or am I missing something here?

Reply:
Yup, you're right, using `to_datetime()` properly is the fastest available option. This discussion can be resolved.